src/RuntimeGeneratedFunctions.jl

```diff
@@ -174,19 +185,25 @@ function _cache_body(cache_tag, id, body)
         # canonical one rather than `body`. This ensures the lifetime of the
         # body in the cache will always cover the lifetime of the parent
         # `RuntimeGeneratedFunction`s when they share the same `id`.
-        cached_body = haskey(cache, id) ? cache[id] : nothing
+        #
+        # Tricky case #2: Unless we hold a separate reference to
+        # `cache[id].value`, the GC can collect it (causing it to become
+        # `nothing`). So root it in a local variable first.
+        #
+        cached_body = haskey(cache, id) ? cache[id].value : nothing
         cached_body = cached_body !== nothing ? cached_body : body
```

Comment:
It would be possible to micro-optimize the code and avoid one lookup:
```suggestion
        cached_body = get(cache, id, nothing)
        cached_body = cached_body !== nothing ? cached_body.value : body
```

Reply:
This code path basically never gets hit (it's only used for compilation and for `show`) so I guess don't care much about performance (the cache is even a `Dict{Any,Any}` :sweat_smile: )  But we may as well I guess.